src/fides/api/schemas/connection_configuration/connection_secrets.py

```diff
@@ -12,6 +12,8 @@
 class ConnectionConfigSecretsSchema(BaseModel, abc.ABC):
     """Abstract Base Schema for updating Connection Configuration Secrets"""
 
+    # NOTE: any fields not listed in `_required_components` must also have an
+    # annotated Optional type, in order to be treated effectively as optional fields
     _required_components: List[str]
```

Comment:
ah these `_required_components` is confusing at this point, thanks for improving with the docstring,   This was the original intent: https://github.com/ethyca/solon/blame/085ace7d88ceee0e1de3a4b2105ce68e54e5e431/src/fidesops/schemas/connection_configuration/connection_secrets.py#L28